contracts/contracts/sbtc-registry.clar

```diff
@@ -138,6 +157,29 @@
   )
 )
 
+;; Rotate the signer set, multi-sig principal, & aggregate pubkey
+;; This function can only be called by the bootstrap-signers contract.
+(define-public (rotate-keys (new-keys (list 15 (buff 32))) (new-address principal) (new-aggregate-pubkey (buff 32)))
+  (begin
+    ;; Check that caller is protocol contract
+    (try! (validate-caller))
+    ;; Check that the aggregate pubkey is not already in the map
+    (asserts! (is-none (map-get? aggregate-pubkeys new-aggregate-pubkey)) ERR_AGG_PUBKEY_REPLAY)
+    ;; Check that the new address (multi-sig) is not already in the map
+    (asserts! (is-none (map-get? multi-sig-address new-address)) ERR_MULTI_SIG_REPLAY)
```

Comment:
This can be converted to:

```clarity
(asserts! (map-insert aggregate-pubkeys new-aggregate-pubkey) ERR_AGG_PUBKEY_REPLAY)
```

And then we don't need two lines for both checking replay and setting the value

Reply:
Ah, shoot, yeah, overlooked that, thanks.

Reply:
Completed.